// kelas/index.js
// module.exports.ubah = daysInMonth();
module.exports = {
  parseInputToSemester : function(data){
    return parseInt(data.charAt(0));
  },
  parseInputToYear : function(data){
    return parseInt(data.substring(1, 5));
  },
  daysInMonth : function(month,year) {
      return new Date(year, month, 0).getDate();
  },
  getFullYear: function (){
    var d = new Date();
    return d.getFullYear();
  },
  bulanIni : function(){
    var d = new Date();
    var month = new Array();
    month[0] = "Januari";
    month[1] = "Februari";
    month[2] = "Maret";
    month[3] = "April";
    month[4] = "Mei";
    month[5] = "Juni";
    month[6] = "Juli";
    month[7] = "Agustus";
    month[8] = "September";
    month[9] = "Oktober";
    month[10] = "November";
    month[11] = "Desember";
    return month[d.getMonth()];
  },
  getMonth : function(month){
    switch (month) {
      case "Januari":
        return 1;
        break;
      case "Februari":
        return 2;
        break;
      case "Maret":
        return 3;
        break;
      case "April":
        return 4;
        break;
      case "Mei":
        return 5;
        break;
      case "Juni":
        return 6;
        break;
      case "Juli":
        return 7;
        break;
      case "Agustus":
        return 8;
        break;
      case "September":
        return 9;
        break;
      case "Oktober":
        return 10;
        break;
      case "November":
        return 11;
        break;
      case "Desember":
        return 12;
        break;
    }
  },
  parseKodeToSemester : function(data){
    var smt = data.substring(0, 1);
    if(parseInt(smt)==2){
      var smt_text = "Genap";
    }else if(parseInt(smt)==1){
      var smt_text = "Ganjil";
    }
    var year = data.substring(1, 5);
    return {
      data : data,
      semester_txt : smt_text,
      tahun : year
    }
  },
  fullDateText : function(){
    var date = new Date();

    var day = date.getDate();
    return day+" "+this.bulanIni()+" "+date.getFullYear();
  }
}
